### PR TITLE
Add apple silicon and update deps

### PR DESCRIPTION
**Pull request by DeMoorJasper**
Close T-1099

An attempt to add apple silicon and updated some basic deps.

### CODE REVIEW

File: azure-pipelines-template.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
     displayName: Install dependencies
   - script: npm test
     displayName: Run tests
-  - script: npm run prebuild
+  - script: npm run prebuild --arch=$(parameters.arch)
```
Comment (devongovett) [inline azure-pipelines-template.yml:42]: I see "line 1: parameters.arch: command not found" in the build...

```suggestion
  - script: npm run prebuild --arch=${{ parameters.arch }}
```

**Comment by devongovett** [comment]
@DeMoorJasper so I tried this out. After fixing it to actually build, I get the following error when running it:

```
/Users/devongovett/dev/watcher/prebuilds/darwin-arm64/node.napi.glibc.node: mach-o, but wrong architecture
```

looks like it's still built for x64...

```
$ file prebuilds/darwin-arm64/node.napi.glibc.node
prebuilds/darwin-arm64/node.napi.glibc.node: Mach-O 64-bit bundle x86_64
```

**Comment by DeMoorJasper** [comment]
@devongovett oh in that case I have no clue how we can fix this without actually running the build on a m1 mac...

**Comment by devongovett** [comment]
Appears to be working! I verified by downloading the artifacts from the build and running the tests on an M1 MacBook Air.

Just need to add yarn to the CentOS docker image and I think we'll be good!

**Pull request merged**